kale/utils/metadata_utils.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from kale.utils.pod_utils import is_workspace_dir
+
+METADATA_REQUIRED_KEYS = [
+    'experiment_name',
+    'pipeline_name',
+]
+kale_step_name_regex = r'^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
+kale_name_msg = ("must consist of lower case alphanumeric characters "
+                 "or '-', and must start and end with an alphanumeric"
+                 " character.")
+k8s_valid_name_regex = r'^[\.\-a-z0-9]+$'
+k8s_name_msg = ("must consist of lower case alphanumeric characters, "
+                "'-' or '.'")
+volume_types = ['pv', 'pvc', 'new_pvc', 'clone']
+volume_required_fields = ['name', 'annotations', 'size', 'type', 'mount_point']
+
+
+def validate_metadata(metadata):
```

Comment:
Should this maybe be renamed to `parse_metadata`, since it modifies its argument? Validation functions are usually expected to perform only validation.

Also, you may want to avoid modifying the argument altogether by deepcopying the provided metadata object before performing any change. Performance is probably not an issue here, and its a bit safer since you can avoid aliasing bugs.